core-plugins/src/test/java/co/cask/hydrator/plugin/transform/PythonEvaluatorTest.java

```diff
@@ -204,12 +204,37 @@ public void testSchemaValidation() throws Exception {
                                                                    ImmutableMap.<String, Object>of(
                                                                      CoreValidator.ID, new CoreValidator()));
     new PythonEvaluator(config).configurePipeline(configurer);
+  }
+
+  @Test
+  public void testSchemaValidation() throws Exception {
+    Schema outputSchema = Schema.recordOf(
+      "smallerSchema",
+      Schema.Field.of("intField", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("longField", Schema.of(Schema.Type.LONG)));
+
+    PythonEvaluator.Config config = new PythonEvaluator.Config(
+      "def transform(x, emitter, context):\n" +
+        " y = {}\n" +
+        " y['intField'] *= 1024\n" +
+        " y['longField'] *= 1024\n" +
+        " emitter.emit(y)",
+      outputSchema.toString());
```

Comment:
I'm thinking that `y` needs to be a shallow copy of `x`. But if you look at the docs for the plugin (http://docs.cask.co/cdap/4.1.1/en/developers-manual/pipelines/plugins/transforms/pythonevaluator-transform.html) you will see this:
```
'def transform(record, emitter, context): record['count'] *= 1024; emitter.emit(record)'
```
which operates on the input record directly.